operator/anarchyruntime.py

```diff
@@ -28,7 +28,7 @@ def __init__(
         self.pod = self.core_v1_api.read_namespaced_pod(self.pod_name, self.operator_namespace)
         self.running_all_in_one = '' != os.environ.get('ODO_S2I_SCRIPTS_URL', '')
         if self.running_all_in_one:
-            self.anarchy_service_name = os.environ.get('ANARCHY_SERVICE', socket.gethostbyname(os.environ.get('HOSTNAME')))
+            self.anarchy_service_name = os.environ.get('ANARCHY_SERVICE', os.environ.get('HOSTNAME'))
```

Comment:
The anarchy service should simply be the pod name, which is resolvable from within an all-in-one pod such as running in odo.